api/handlers.go

```diff
@@ -147,502 +147,504 @@ func (si *ServerImplementation) MakeHealthCheck(ctx echo.Context) error {
+	if errors.As(err, &perr) {
+		return badRequest(ctx, perr.Error())
+	} else if errors.Is(err, ErrVerifyFailedEndpoint) {
+		return badRequest(ctx, err.Error())
 	}
+
+	return nil
+}
+
+func (si *ServerImplementation) verifyHandler(operationID string, ctx echo.Context) error {
+	return Verify(si.disabledParams, operationID, ctx, si.log)
 }
 
 // LookupAccountByID queries indexer for a given account.
 // (GET /v2/accounts/{account-id})
 func (si *ServerImplementation) LookupAccountByID(ctx echo.Context, accountID string, params generated.LookupAccountByIDParams) error {
-	return si.verifyHandler("LookupAccountByID", ctx, func() error {
+	if err := si.verifyHandler("LookupAccountByID", ctx); err != nil {
+		return si.handleVerifyError(err, ctx)
+	}
 
-		addr, errors := decodeAddress(&accountID, "account-id", make([]string, 0))
-		if len(errors) != 0 {
-			return badRequest(ctx, errors[0])
-		}
+	addr, errors := decodeAddress(&accountID, "account-id", make([]string, 0))
+	if len(errors) != 0 {
+		return badRequest(ctx, errors[0])
+	}
 
-		options := idb.AccountQueryOptions{
-			EqualToAddress:       addr[:],
-			IncludeAssetHoldings: true,
-			IncludeAssetParams:   true,
-			Limit:                1,
-			IncludeDeleted:       boolOrDefault(params.IncludeAll),
-		}
+	options := idb.AccountQueryOptions{
+		EqualToAddress:       addr[:],
+		IncludeAssetHoldings: true,
+		IncludeAssetParams:   true,
+		Limit:                1,
+		IncludeDeleted:       boolOrDefault(params.IncludeAll),
+	}
 
-		accounts, round, err := si.fetchAccounts(ctx.Request().Context(), options, params.Round)
-		if err != nil {
-			return indexerError(ctx, fmt.Errorf("%s: %w", errFailedSearchingAccount, err))
-		}
+	accounts, round, err := si.fetchAccounts(ctx.Request().Context(), options, params.Round)
+	if err != nil {
+		return indexerError(ctx, fmt.Errorf("%s: %w", errFailedSearchingAccount, err))
+	}
 
-		if len(accounts) == 0 {
-			return notFound(ctx, fmt.Sprintf("%s: %s", errNoAccountsFound, accountID))
-		}
+	if len(accounts) == 0 {
+		return notFound(ctx, fmt.Sprintf("%s: %s", errNoAccountsFound, accountID))
+	}
 
-		if len(accounts) > 1 {
-			return indexerError(ctx, fmt.Errorf("%s: %s", errMultipleAccounts, accountID))
-		}
+	if len(accounts) > 1 {
+		return indexerError(ctx, fmt.Errorf("%s: %s", errMultipleAccounts, accountID))
+	}
 
-		return ctx.JSON(http.StatusOK, generated.AccountResponse{
-			CurrentRound: round,
-			Account:      accounts[0],
-		})
-	})()
+	return ctx.JSON(http.StatusOK, generated.AccountResponse{
+		CurrentRound: round,
+		Account:      accounts[0],
+	})
 }
 
 // SearchForAccounts returns accounts matching the provided parameters
 // (GET /v2/accounts)
 func (si *ServerImplementation) SearchForAccounts(ctx echo.Context, params generated.SearchForAccountsParams) error {
-	return si.verifyHandler("SearchForAccounts", ctx, func() error {
+	if err := si.verifyHandler("SearchForAccounts", ctx); err != nil {
+		return si.handleVerifyError(err, ctx)
```

Comment:
I don't think this is needed anymore, you can use `return badRequest(ctx, err.Error())`

Reply:
Done